File: Assets/Bundles/Client/Resources/TS/Signals/BlockPlaceClientSignal.lua
-- Compiled with unity-ts v2.1.0-75
local BlockPlaceClientSignal
do
	BlockPlaceClientSignal = setmetatable({}, {
		__tostring = function()
			return "BlockPlaceClientSignal"
		end,
	})
	BlockPlaceClientSignal.__index = BlockPlaceClientSignal
	function BlockPlaceClientSignal.new(...)
		local self = setmetatable({}, BlockPlaceClientSignal)
		return self:constructor(...) or self
	end
	function BlockPlaceClientSignal:constructor(pos, block, placer)
		self.pos = pos
		self.block = block
		self.placer = placer
	end
end
return {
	BlockPlaceClientSignal = BlockPlaceClientSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Client/Signals/BlockPlaceClientSignal.ts
import { Entity } from "Shared/Entity/Entity";
import { Block } from "Shared/VoxelWorld/Block";

export class BlockPlaceClientSignal {
	constructor(
		public readonly pos: Vector3,
		public readonly block: Block,
		public readonly placer: Entity | undefined,
	) {}
}

export class BlockGroupPlaceClientSignal {
	constructor(
		public readonly pos: Vector3[],
		public readonly block: Block[],
		public readonly placer: Entity | undefined,
	) {}
}
